tools/kagami/README.md

```diff
@@ -1,94 +1,139 @@
 # Kagami (Teacher and Exemplar and/or Looking glass)
 
-Generate and validate the automatically generated data files shipped with Iroha.
+Kagami is a tool used to generate and validate automatically generated data files that are shipped with Iroha.
 
-### Building
+## Build
 
-Use
+From anywhere in the repository, run:
 
 ```bash
 cargo build --bin kagami
 ```
 
-anywhere in this repository. This will place `kagami` inside the `target/debug/` directory (from the root of the repository).
+This will place `kagami` inside the `target/debug/` directory (from the root of the repository).
 
-### Usage
+## Usage
 
-```bash
-kagami 2.0.0-pre-rc.3
-Soramitsu Iroha2 team (https://github.com/orgs/soramitsu/teams/iroha2)
-Tool generating the cryptorgraphic key pairs,
+Run Kagami:
 
-USAGE:
+```
 kagami <SUBCOMMAND>
+```
 
-OPTIONS:
--h, --help       Print help information
--V, --version    Print version information
+### Subcommands
 
-SUBCOMMANDS:
-crypto     Generate cryptorgraphic key pairs
-docs       Generate a Markdown reference of configuration parameters
-genesis    Generate a default genesis block that is used in tests
-help       Print this message or the help of the given subcommand(s)
-schema     Generate schema used for code generation in Iroha SDKs
-```
+|        Command        |                             Description                              |
+| --------------------- | -------------------------------------------------------------------- |
+| [`crypto`](#crypto)   | Generate cryptographic key pairs                                     |
+| [`docs`](#docs)       | Generate a Markdown reference of configuration parameters            |
+| [`genesis`](#genesis) | Generate the default genesis block that is used in tests             |
+| [`schema`](#schema)   | Generate the schema used for code generation in Iroha SDKs           |
+| [`tokens`](#tokens)   | Generate a list of predefined permission tokens and their parameters |
+| `help`                | Print the help message for the tool or a subcommand                  |
 
-#### Key generation
+## `crypto`
 
-With a few examples.
+The `crypto` command generate cryptographic key pairs using the given algorithm and either private key or seed.
 
-```bash
-$ ./kagami crypto
-```
+|     Option      |                                         Description                                          | Default value  |  Type  |
+| --------------- | -------------------------------------------------------------------------------------------- | -------------- | ------ |
+| `--algorithm`   | The algorithm used to generate the key-pair: `Ed25519`, `Secp256k1`, `BlsNormal`, `BlsSmall` | `Ed25519`      | String |
```

Comment:
```suggestion
| `--algorithm`   | The algorithm used to generate the key-pair: `ed25519`, `secp256k1`, `bls_normal`, `bls_small` | `ed25519`      | String |
```

FIY you can look at [`FromStr` implementation](https://github.com/hyperledger/iroha/blob/54332fb23e9adf455d21794737a5928d971be976/crypto/src/lib.rs#L86) to see how CLI parse from input to Rust data structure